### ISSUE TITLE
Don't warn about double ping with !!tell @user

### ISSUE THREAD

**Issue by MadaraUchiha** [Zirak/SO-ChatBot#236]
Instead, just perform the command normally, and don't ping the user.

> Me: !!tell @Zirak ead
> Bot: Zirak: Eat A Dick™

The rationale being that with the current warning, the target would still get double pinged (one from me, and one from bot after I edit my message to contain no pings), plus I get pinged about the double ping.

**Comment by xbenjii** [comment]
`!!tell Zirak ead` works too.

**Comment by MadaraUchiha** [comment]
@xbenjii Yes, but !!tell @som and having it complete to !!tell @SomeSuperComplicatedNicknameWhichIsAHundredCharactersLong is a lot more convenient.

**Comment by awalGarg** [comment]
:+1: Typing out `!!tell longUsernameWithoutTabCompletion ...` is weird, and doing tab completion and removing the `@` later is messy.

Only issue with this could be when the bot is running for a long time, thus acting slowly. A newbie might get confused if a rail of messages follow in between, he might miss the message by the bot in a flood of messages.

**Comment by MadaraUchiha** [comment]
@awalGarg The bot will still mention him by name, just not ping.

**Comment by Zirak** [comment]
That was the original behaviour, I changed it so the frequency of double pings will decrease.

If I were to `!!tell @MadaraUchiha ead` successfully, I'd do it again in the future. In this scheme, you'll be annoyed, and less likely to repeat.

**Comment by MadaraUchiha** [comment]
Yeah, but the the bot wouldn't ping me again, so what do I care?

**Comment by Zirak** [comment]
Over time and overall you'll have less double pings. You remember that the bot does this annoying thing and you won't do it, because if you do the command won't work and you'll be reminded.

**Comment by MadaraUchiha** [comment]
How will you have less double pings? With the current behavior, you always either get 1 or 2 pings, with my suggested behavior you always only get 1 ping.

**Comment by Zirak** [comment]
*Scene: Coffee shop, mid-afternoon. It's crowded, the air abuzz with conversation and laughter. Waitresses bustle about, balancing trays of food and drink, taking orders and looking more tired by the moment. Enter @MadaraUchiha, stage right. He approaches the counter, intending to order a double Bananalicious.*

Madara *(shouting)*: TOP OF THE MORNIN' TO YOU, M'LADY. MAY I ACQUIRE SOME NOURISHMENT IN THE FORM OF BANANA DELICIOUSNESS?

Cashier/waitress: Uuuhhh, sure *(whispers: weirdo)*. One Bananalicious!

Madara: MOST EXQUISITE. THANK YOU!

*Scene: The following day. Same hustle bustle. A hipster is using a typewriter in the corner. Everyone around him glances and sighs at him every now and then. Enter @MadaraUchiha, stage right. He approaches the counter, intending to order a Peachstrocity.*

Madara *(shouting)*: GUTTEN MORNIN', FAIR MAIDEN. IS IT WITHIN THE REALM OF POSSIBILITY THAT A DRINK MADE OF MONSTROUS PEACHES ENTER MY BOWELS?

*Cashier/waitress thinks to herself: This guy, every day...if only there was a way he could stop shouting!*

*Suddenly, a wormhole appears, enveloping the world around her. For visual effects, refer to https://youtu.be/tNFs1swHw5I?t=70*

*The cashier/waitress enters a new dimension, where she is more assertive*

Cashier/waitress: Sir, can you please stop shouting?

Madara: Oh dear lord, my apologies! However did I not notice such a thing could annoy my fellow members of society? I shall cease at once, and transcend to better humanhood!

*And so, our hero learned a valuable lesson. He stopped shouting at everyone, was less harassed in general, caught AIDS and died.*

*** FIN***

-------

***tl;dr*** The bot says that double pinging is bad. In the current invocation the recipient will get a double ping, but the bot teaches you that `/tell @someone something` doesn't work, so next time you use `/tell`, you'll drop the @, therefore reducing the overall amount of double pings.

**Comment by AmaanC** [comment]
Haha, nice story. But with the method Madara proposes, you don't get *any*
double pings. Examples:

!!tell Zirak abc
@Zirak Lorem ipsum

!!tell @Zirak ABC
Zirak: Lorem ipsum

In both situations, the user is only pinged once.

**Comment by Ralt** [comment]
@AmaanC the bot doesn't ping the user though. The user's attention is attracted to the ping, not to the bot's message. Let's not forget that the audience of this command are zombie-like minded people who like shiny stuff.

By the way, SO chat has a nice feature (cc @NickDugger) that lets you specify only the 3 first letters of the nickname to ping someone, so you don't have to write the full name of the person you want to ping.

**Comment by Zirak** [comment]
Oh, I completely missed that detail (the bot's reply not being a ping). Whoops.

Not sure how great that is, especially to first time botters in crowded times like we have:

Zirak: !!tell @ThatNewGuy google something
MadaraUchiha: Java is my favourite language. The JVM is a terrific platform.
AwalGarg: omigod ubuntu
Bot: ThatNewGuy: Here's the fucking google results
SomeGuy: I'M READING BOOKS WHILE EATING STAKE
Ralt: Isn't lisp amazing? I just orgasmed.

ThatNewGuy is less likely to see the results, just my ping telling him to google something. Which is pseudo fine. But eh.

**Comment by AmaanC** [comment]
Yeah, that's a fair point. I just wanted to clear up that there wouldn't be
double pings. I think what we have now is good enough. With the "feature"
Florian mentioned, it should be alright to use in general.

**Comment by SomeKittens** [comment]
Eating.... stake?

![](http://img2.wikia.nocookie.net/__cb20130603182504/horrorclix/images/9/9c/Buffy_stake.gif)

**Comment by AmaanC** [comment]
Yeah, Zirak doesn't know the difference between a stake and steak. Suck it, @Zirak.

**Comment by MadaraUchiha** [comment]
Dude, seriously, the current method leaves a chance for double pings, what I suggest will completely eliminate double pings... Either you ping, or the bot pings, there are no double pings, at any time.

Typing long-ass names without completion is hard, and even the @ Z I R TAB CTRL+LEFT-ARROW DELETE END combination is clumsy and annoying. I *still* don't see your argument, with or without the fancy story.

**Comment by MadaraUchiha** [comment]
As for the user missing what was told to him, if you (as someone who uses !!google at that specific time) seriously cares about whether or not the target user sees the result, or when you see the room is very busy, you can revert to the old method, it will still work.

Also, someone who's paying attention to what's going on will still see messages where he's mentioned by name, even without the ping.

**Comment by MadaraUchiha** [comment]
Bump. Do we want this or not?

**Comment by awalGarg** [comment]
This can be implemented in addition to the present behavior. If Cap sees an `@` in front of the name received, she can send a message without an `@`, else, she can do whatever she is doing right now.

This'd not break BC, more flexibility, <insert unrelated good point here>.

**Comment by MadaraUchiha** [comment]
That's exactly what I'm suggesting for the past 4-5 messages.

**Comment by awalGarg** [comment]
:+2:

**Comment by Zirak** [comment]
@MadaraUchiha I'm not all too excited about it, but if a nice chunk of people like it then I can give it a try.

**Comment by rlemon** [comment]
+1, Zirak is smelly and should be ignored.
